lib/fog/vsphere/requests/compute/get_vm_first_nvme_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Fog
+  module Vsphere
+    class Compute
+      class Real
+        def get_vm_first_nvme_controller(vm_id)
+          ctrl = get_vm_ref(vm_id).config.hardware.device.find { |device| device.is_a?(RbVmomi::VIM::VirtualNVMEController) }
+          raise Fog::Vsphere::Compute::NotFound, "No NVME controller found for #{vm_id}" unless ctrl
+          {
+            type: ctrl&.class.to_s,
+            device_info: ctrl&.deviceInfo,
+            bus_number: ctrl&.busNumber,
+            key: ctrl&.key
+          }
+        end
+      end
+      class Mock
+        def get_vm_first_nvme_controller(vm_id); end
```

Comment:
The mock class returns `nil`, while the real method will throw an exception and never return `nil`. 
I think it would be nice to have the behavior be equal

Reply:
We are not doing that for the other(scsi and sata) controllers, I dont think it's necessary to do that as it's the mock class.